plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/contentassist/N4JSProposalProvider.xtend

```diff
@@ -160,104 +172,66 @@ class N4JSProposalProvider extends AbstractN4JSProposalProvider {
+	}
+
+	def protected getStyledDisplayString(String qualifiedName, String shortName, int version) {
+		val result = new StyledString();
+		val parsedQualifiedName = qualifiedNameConverter.toQualifiedName(qualifiedName);
+		if (parsedQualifiedName.segmentCount > 1) {
+			val dashName = ' - ' + qualifiedNameConverter.toString(parsedQualifiedName.skipLast(1));
+			val lastSegment = parsedQualifiedName.lastSegment;
+			val typeVersion = if (version === 0) "" else N4IDLGlobals.VERSION_SEPARATOR + String.valueOf(version);
+
+			var String combinedLabel;
+			if (shortName.endsWith(lastSegment)) {
+				combinedLabel = lastSegment + typeVersion + dashName;
+			} else {
+				combinedLabel = shortName + dashName + " alias for " + lastSegment + typeVersion;
 			}
-			return stringifier.apply(element, parsedQualifiedName, parsedQualifiedName.lastSegment)
+			result.append(combinedLabel, StyledString.QUALIFIER_STYLER);
 		}
-		return tryGetDisplayString(element, shortName) ?: stringifier.apply(element, qualifiedNameConverter.toQualifiedName(qualifiedName), shortName)
+		return result;
 	}
 
 	override protected getImage(IEObjectDescription description) {
 		val clazz = description.EClass
 		return super.getImage(EcoreUtil.create(clazz))
 	}
 
-	/**
-	 * Overridden to avoid calls to IEObjectDescription#getEObjectOrProxy
-	 *
-	 * Remove before merging to master!
-	 * Due to IDL, the proxy element is retrieved anyway, but used only if it is a TClassifier
-	 * /
-	override protected getStyledDisplayString(IEObjectDescription description) {
-		val element = description.getEObjectOrProxy();
-		val qualifiedName = description.qualifiedName;
-		val shortName = description.name;
-		if (qualifiedName == shortName) {
-			if (shortName.segmentCount >= 1) {
-				return stringifier.apply(element, qualifiedName, shortName.lastSegment)
-			} 
-		}
-		// don't recompute the qualified name again
-		return stringifier.apply(element, qualifiedName, qualifiedNameConverter.toString(shortName))
-	}
-	*/
-
 	/**
 	 * If the element is an instance of {@link TClassifier} this method
 	 * returns a user-faced string description of the version information.
-	 *
+	 * 
 	 * Otherwise, this method returns an empty string.
 	 */
-	private def String getTypeVersionString(EObject element) {
-		if (!element.eIsProxy && element instanceof TClassifier &&
-			(element as TClassifier).declaredVersion != 0) {
-			return N4IDLGlobals.VERSION_SEPARATOR + Integer.toString((element as TClassifier).declaredVersion)
-		}
-		return "";
-	}
-
-	/**
-	 * Returns the display string for a non-proxy element, otherwise null.
-	 */
-	private def tryGetDisplayString(EObject element, String shortName) {
-		if (!element.eIsProxy && element instanceof Type) {
-			val qualifiedTypeName = qualifiedNameProvider.getFullyQualifiedName(element)
-			if (qualifiedTypeName !== null) {
-				return stringifier.apply(element, qualifiedTypeName, shortName)
-			}
+	private def int getTypeVersionString(EObject element) {
```

Comment:
getTypeVersionString should either return a String or be renamed. See JavaDoc, too.